src/main/scala/bi/deep/DruidSchemaReader.scala

```diff
@@ -12,21 +12,22 @@ import scala.collection.immutable.ListMap
 
 
 case class DruidSchemaReader(config: Config) extends DruidSegmentReader {
-  def calculateSchema(files: Seq[Segment])(implicit fs: FileSystem): SparkSchema = {
-    val druidSchemas = files.map { file =>
+  def calculateSparkSchema(files: Seq[Segment])(implicit fs: FileSystem): SparkSchema = {
+    var druidSchemas = files.map { file =>
       withSegment(file.path.toString, config) { segmentDir =>
-        val qi = indexIO.loadIndex(segmentDir)
-        val qisa = new QueryableIndexStorageAdapter(qi)
-        DruidSchemaReader.readDruidSchema(qisa)
+        val queryableIndex = indexIO.loadIndex(segmentDir)
+        val queryableIndexStorageAdapter = new QueryableIndexStorageAdapter(queryableIndex)
+        DruidSchemaReader.readDruidSchema(queryableIndexStorageAdapter)
       }
     }
 
-    druidSchemas.foreach(druidSchema => druidSchema.validateFields())
+    druidSchemas = druidSchemas.map(druidSchema => druidSchema.validateFields())
```

Comment:
No need for `var druidSchema` it is a bad practice to use `var` if it is uneccesary, we can chain the next map together. Then again the `validateFileds` is only used here because the object was not created properly in the first place. Addressing previous comment will also fix this one buy removing line 24 completrly